### ISSUE TITLE
Diagnostic actions registered using RegisterCodeBlockAction do not run in IDE

### ISSUE THREAD

**Issue by mattwar** [dotnet/roslyn#1709]
using 

```C#
context.RegisterCodeBlockAction(MyHandler);
```

results in no execution during IDE diagnostic analysis.

However, the start/end pattern using

```C#
context.RegisterCodeBlockStartAction(c => { c.RegisterCodeBlockEndAction(MyHandler); }
```

does cause the action to run.

**Comment by mavasani** [comment]
@mattwar I am unable to repro this issue, I have also added compiler/IDE tests for code block actions to confirm that these are triggered ( see #1827 )

Probably I need more context to repro this:
1. Were you seeing the issue only during method body editing? IDE has a special code path to optimize for those scenarios.
2. Were you able to set a breakpoint and verify that Start/End pattern actions were getting executed, but not the CodeBlockAction ones?
3. Can you share your analyzer code + test code on which analyzer was being execute?

**Comment by mattwar** [comment]
I can still repro it. 

Yes, I can set break points and stop at Start/End actions, but not on CodeBlock action directly.  If I register as code block action then the analyzer does not run and I don't get diagnostics, but if I switch to start/end then my analyzer is run and I do get diagnostics.

My analyzer is here:  https://github.com/mattwar/nullaby/blob/master/src/Nullaby/Analyzer/NullAnalyzer.cs#L67

You can see in the source for analyzer that I have commented out registering for CodeBlock and have it call the CodeBlockStart + CodeBlockEnd.

**Comment by mavasani** [comment]
@mattwar Finally got it to repro. The issue doesn't repro if you register both the CodeBlockAction and CodeBlockStartAction, the IDE runs both the actions for that case (as in the added tests). If you register just the CodeBlockAction, IDE is skipping it just as you indicated. I'll work on a fix. Thanks!

**Comment by mavasani** [comment]
@mattwar This should be fixed now, I verified with your analyzer as well.
